### ISSUE TITLE
Allow different line widths in contour

### ISSUE THREAD

**Issue by alex-robinson** [MakieOrg/Makie.jl#3910]
### Feature description
In the same way that different colors can be used for different contour lines when calling the function `contour`, it would be very valuable if the same flexibility could be extended to line widths.

Of course, this can be achieved with multiple, separate calls to `contour`, but this makes scripts longer and more complicated especially when, e.g., building animations with the `record` function, as it forces you to keep track of multiple contour objects. Thus including this feature would help a lot!

A MWE is shown below, where the last case fails with an error when multiple linewidths are prescribed:

```julia
using CairoMakie

x = -10:0.1:10;
y = -10:0.1:10;
z = x' .* y;

# One color and one linewidth for all contours, works:
begin
    levels = [-50,-10,10,50];
    color = :red;
    linewidth = 1;
    contour(x,y,z;levels=levels,color=color,linewidth=linewidth)
end

# Multiple colors and one linewidth for contours, works:
begin
    levels = [-50,-10,10,50];
    n = length(levels);
    colors = fill(:red,n);
    colors[2:3] .= :green;
    linewidth = 1;
    contour(x,y,z;levels=levels,color=colors,linewidth=linewidth)
end

# Multiple colors and multiple linewidths for contours, doesn't work:
begin
    levels = [-50,-10,10,50];
    n = length(levels);
    colors = fill(:red,n);
    colors[2:3] .= :green;
    linewidths = fill(1,n);
    linewidths[2:3] .= 3;
    contour(x,y,z;levels=levels,color=colors,linewidth=linewidths)
end
```

**Comment by asinghvi17** [comment]
hmm, are you proposing that all line attributes be per-level (including linestyle, glow, etc)?  That would be pretty cool, I'd definitely love to see a PR for this.

One thing to keep in mind is that assigning these attributes should always require a manual `levels` specification - otherwise, there's no way for the user to know the number of levels in the plot.

**Comment by alex-robinson** [comment]
Yes, I didn't extend my thinking beyond `linewidth`, but indeed, my intuition when using `contour` was that it would work that way for all line attributes. 

I agree, it does need the number `levels` specified. And some more informative errors in that case would be helpful too then. 

Unfortunately I'm not one yet at the level to contribute code to this change, but I would be very happy to see it implemented.
